pkg/operator/controller/dns_status_test.go

```diff
@@ -0,0 +1,354 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDNSStatusConditions(t *testing.T) {
+	testCases := []struct {
+		description                                string
+		clusterIP                                  string
+		avail, desire                              int32
+		degradeStatus, progressStatus, availStatus operatorv1.ConditionStatus
+	}{
+		{
+			description:    "no cluster ip, 0/0 pods available",
+			clusterIP:      "",
+			avail:          0,
+			desire:         0,
+			degradeStatus:  operatorv1.ConditionTrue,
+			progressStatus: operatorv1.ConditionFalse,
+			availStatus:    operatorv1.ConditionFalse,
```

Comment:
As @ironcladlou suggested on the ingress pr https://github.com/openshift/cluster-ingress-operator/pull/200 , this can be simplified:
```
type conditions struct {
	degraded, progressing, available bool
}
And then in the test case, we could do
...
expectedConditions: conditions{true, false, false}
...
```